packages/material-ui/src/Stepper/Stepper.test.js

```diff
@@ -37,143 +35,184 @@ describe('<Stepper />', () => {
-      const wrapper = shallow(
-        <Stepper linear={false} activeStep={0}>
-          <div className="child-0" />
-          <div className="child-1" />
-          <div className="child-2" />
+      const { container, setProps } = render(
+        <Stepper nonLinear activeStep={0}>
+          <Step />
+          <Step />
+          <Step />
         </Stepper>,
       );
-      expect(wrapper.find('.child-0').props().active).to.equal(true);
-      expect(wrapper.find('.child-1').props().active).to.equal(false);
-      expect(wrapper.find('.child-2').props().active).to.equal(false);
-      wrapper.setProps({ activeStep: 1 });
-      expect(wrapper.find('.child-0').props().active).to.equal(false);
-      expect(wrapper.find('.child-1').props().active).to.equal(true);
-      expect(wrapper.find('.child-2').props().active).to.equal(false);
-      wrapper.setProps({ activeStep: 2 });
-      expect(wrapper.find('.child-0').props().active).to.equal(false);
-      expect(wrapper.find('.child-1').props().active).to.equal(false);
-      expect(wrapper.find('.child-2').props().active).to.equal(true);
+
+      const steps = container.querySelectorAll('.MuiStep-root');
+      const connectors = container.querySelectorAll('.MuiStepConnector-root');
+
+      expect(steps[0]).to.not.have.class('MuiStep-completed');
+      expect(steps[1]).to.not.have.class('MuiStep-completed');
+      expect(steps[2]).to.not.have.class('MuiStep-completed');
+      expect(connectors[0]).not.to.have.class('Mui-disabled');
+      expect(connectors[1]).not.to.have.class('Mui-disabled');
+
+      setProps({ activeStep: 1 });
+
+      expect(steps[0]).to.not.have.class('MuiStep-completed');
+      expect(steps[1]).to.not.have.class('MuiStep-completed');
+      expect(steps[2]).to.not.have.class('MuiStep-completed');
+      expect(connectors[0]).to.not.have.class('Mui-disabled');
+      expect(connectors[0]).to.have.class('MuiStepConnector-active');
+      expect(connectors[1]).not.to.have.class('Mui-disabled');
+
+      setProps({ activeStep: 2 });
+
+      expect(steps[0]).to.not.have.class('MuiStep-completed');
+      expect(steps[1]).to.not.have.class('MuiStep-completed');
+      expect(steps[2]).to.not.have.class('MuiStep-completed');
+      expect(connectors[0]).to.not.have.class('Mui-disabled');
+      expect(connectors[1]).to.not.have.class('Mui-disabled');
+      expect(connectors[1]).to.have.class('MuiStepConnector-active');
     });
 
     it('passes index down correctly when rendering children containing arrays', () => {
-      const wrapper = shallow(
-        <Stepper linear={false}>
-          <div />
-          {[<div key={1} />, <div key={2} />]}
-        </Stepper>,
-      );
-
-      const steps = wrapper.children().find('div');
-      expect(steps.at(0).props().index).to.equal(0);
-      expect(steps.at(1).props().index).to.equal(1);
-      expect(steps.at(2).props().index).to.equal(2);
+      // I don't see how to test it using react-testing-library
```

Comment:
Maybe you can query the rendered text of the step label?

Reply:
```jsx
function CustomStep({ index }) {
  return <div data-index={index} data-testid="step" />;
}

getAllByTestId('step')[0]).to.have.attribute('data-index', "0");
// and so forth
```
?

Reply:
Done. Thanks for the tip guys!